### ISSUE TITLE
TN3270 - ZWED0097I Websocket closed

### ISSUE THREAD

**Issue by Martin-Zeithaml** [zowe/zlux#1021]
The terminal is not working after the second usage:
* The first attempt to use TN3270 is fine
* The second attempt shows only black screen

The log shows this suspicious messages.
```
ZWED0275I - Host=example.com, Port=23, ClientIP=256.257.258.1 Writing to host socket. Length=170
ZWED0136I - Host=example.com, Port=23, ClientIP=256.257.258.1 Content to be sent to host socket=
�
ZWED0097I - Host=example.com, Port=23, ClientIP=256.257.258.1 Websocket closed. Total remaining terminals connected: 0
```
The buffer length should be 170 bytes, but actually is only `\xEF \xBF \xBD`.

**Comment by 1000TurquoisePogs** [comment]
Can you reproduce this such that with `message` logging on you can see what the terminal in the browser sends?
https://github.com/zowe/tn3270-ng2/wiki/Loggers#developingdebugging
I'm curious to see if the terminal protocol is failing and therefore sending nonsense, or if the network code is failing instead.

**Comment by 1000TurquoisePogs** [comment]
Sadly, the "content to be sent to host socket" is not a hex printout. It's just the raw bytes.
So, if you could read that with HEX ON in ISPF, you'd see the actual hex bytes.

**Comment by 1000TurquoisePogs** [comment]
Just a thought... if the message sent resulted in a socket close, it could be a handshake failure.
Basically, if the client says it supports X,Y,Z and the server only supports A,B,C, then the server has no choice but to close the socket because they're incompatible.
Or worse. the client could be trying to do a TLS handshake while the server is just telnet without TLS.
So, I wonder about that... is the config wrong such that there's just a total handshake failure

**Comment by Martin-Zeithaml** [comment]
The user is able to use terminal once, sometimes twice normally, it means the handshake was successful for these cases.

**Comment by Martin-Zeithaml** [comment]
I have noticed this:
```
socketRead: tlsRead failed with status: 437 (Connection closed)                                     
HTTP desiredBytes = 65536 bytesRead=-1, so a problem or no more available rc=0x0, reason=0x0. conversation=0x12345678
```
Any idea?

**Comment by 1000TurquoisePogs** [comment]
TN3270 app contacts zss for this discovery feature where zss can tell the terminal a little about the context of what page its on.
This was intended to be able to do advanced app-to-app actions, such as being able to right-click a dataset in the 3.4 menu, and open it in the editor...
But, nobody completed that code, so it doesnt do very much. You can ignore the zss connections.


Just a reminder that i found that NODE_DEBUG=stream will print the bytes that are handed to app-server, including websocket, so it could be used to detect if networking is OK

**Comment by 1000TurquoisePogs** [comment]
When I turned on NODE_DEBUG=stream for a user who had an issue similar to what you describe, and had gateway debug=true, I noticed that when the terminal didnt work, gateway said it was sending the websocket bytes that I expected, but node never said they were received.
What does this mean?
* error after the gateway log message that ended up preventing the send?
* error before node prints its message?
* something in between the two servers drops the bytes?

But, it at least doesn't mean that there's something wrong with express, express-ws, ws, or our websocket terminal logic. Because, they never get the bytes with which to process.

Curious if you see the same, or something different.
